pennylane/transforms/dynamic_one_shot.py

```diff
@@ -246,20 +248,27 @@ def measurement_with_no_shots(measurement):
             else np.nan
         )
 
-    interface = interface or qml.math.get_deep_interface(circuit.data)
+    interface = qml.math.get_deep_interface(results)
     interface = "numpy" if interface == "builtins" else interface
+    interface = "tensorflow" if interface == "tf" else interface
     active_qjit = qml.compiler.active()
 
     all_mcms = [op for op in aux_tapes[0].operations if is_mcm(op)]
     n_mcms = len(all_mcms)
-    mcm_samples = qml.math.hstack(tuple(res.reshape((-1, 1)) for res in results[-n_mcms:]))
+    mcm_samples = qml.math.hstack(
+        tuple(qml.math.reshape(res, (-1, 1)) for res in results[-n_mcms:])
+    )
     mcm_samples = qml.math.array(mcm_samples, like=interface)
     # Can't use boolean dtype array with tf, hence why conditionally setting items to 0 or 1
     has_postselect = qml.math.array(
-        [[int(op.postselect is not None) for op in all_mcms]], like=interface
+        [[int(op.postselect is not None) for op in all_mcms]],
+        like=interface,
+        dtype=mcm_samples.dtype,
```

Comment:
Do we still need the `int` since we specify `dtype`?
```suggestion
        [[op.postselect is not None for op in all_mcms]],
        like=interface,
        dtype=mcm_samples.dtype,
```

Reply:
Good catch